tests/control/ros/test_Conversions.cpp

```diff
@@ -0,0 +1,275 @@
+  EXPECT_EIGEN_EQUAL(make_vector(3.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(4.), values, kTolerance);
+}
+
+TEST_F(ConvertJointTrajectoryTests, QuinticTrajectory)
+{
+  auto quinticTwoWaypointMessage = mTwoWaypointMessage;
+  quinticTwoWaypointMessage.points[0].velocities.assign({3.});
+  quinticTwoWaypointMessage.points[0].accelerations.assign({5.});
+  quinticTwoWaypointMessage.points[1].velocities.assign({4.});
+  quinticTwoWaypointMessage.points[1].accelerations.assign({6.});
+
+  const auto trajectory = convertJointTrajectory(
+    mStateSpace, quinticTwoWaypointMessage);
+
+  ASSERT_TRUE(!!trajectory);
+  ASSERT_DOUBLE_EQ(0., trajectory->getStartTime());
+  ASSERT_DOUBLE_EQ(1., trajectory->getEndTime());
+  EXPECT_DOUBLE_EQ(1., trajectory->getDuration());
+  EXPECT_EQ(1, trajectory->getNumSegments());
+  EXPECT_EQ(5, trajectory->getNumDerivatives());
+
+  auto state = mStateSpace->createState();
+  Eigen::VectorXd values;
+
+  trajectory->evaluate(0., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(1.), values, kTolerance);
+
+  trajectory->evaluate(1., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(2.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(3.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(4.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0., 2, values);
+  EXPECT_EIGEN_EQUAL(make_vector(5.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 2, values);
+  EXPECT_EIGEN_EQUAL(make_vector(6.), values, kTolerance);
+}
+
+TEST_F(ConvertJointTrajectoryTests, LinearTrajectoryWithDifferentOrdering)
+{
+  const auto linearTwoWaypointMessage = mTwoWaypointMessage2Joints;
+  const auto trajectory = convertJointTrajectory(
+    mStateSpace2Joints, linearTwoWaypointMessage);
+
+  ASSERT_TRUE(!!trajectory);
+  ASSERT_DOUBLE_EQ(0., trajectory->getStartTime());
+  ASSERT_DOUBLE_EQ(1., trajectory->getEndTime());
+  EXPECT_DOUBLE_EQ(1., trajectory->getDuration());
+  EXPECT_EQ(1, trajectory->getNumSegments());
+  EXPECT_EQ(1, trajectory->getNumDerivatives());
+
+  auto state = mStateSpace2Joints->createState();
+  Eigen::VectorXd values;
+
+  trajectory->evaluate(0., state);
+  mStateSpace2Joints->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(2., 1.), values, kTolerance);
+
+  trajectory->evaluate(1., state);
+  mStateSpace2Joints->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(3., 2.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0.5, 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(1., 1.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0.5, 2, values);
+  EXPECT_EIGEN_EQUAL(make_vector(0., 0.), values, kTolerance);
+}
```

Comment:
- Add a test for a `MetaSkeleton` that has multiple joints with the same name. This is possible by creating a `Group` that contains the `Joint` with the same name in two different `Skeleton`s.
- Add a test for a `MetaSkeleton` that is missing one of the joints named in the `JointTrajectoryMessage`.